sherpa/astro/tests/test_astro_data2.py

```diff
@@ -1548,13 +1547,145 @@ def test_pha_grouping_changed_filter_1160(make_test_pha):
+    """Test based on work on #1160
+
+    See test_pha_grouping_changed_1160_grped_no_filter
+    """
+
+    pha = make_grouped_pha
+
+    # Can not say
+    #   pha.notice(2, 4)
+    # as we have already done a filter, so this would not
+    # act to ignore the first channel.
+    #
+    # The ignore(lo=5) line is not needed as it is already excluded by
+    # a bad-quality channel, but users can say this, so check the the
+    # response.
+    #
+    pha.ignore(hi=1)
+    pha.ignore(lo=5)
+
+    # Dropping channel 1 means the first group gets dropped, so we
+    # only have channel 4 left.
+    #
+    assert pha.get_filter() == "4"
+
+    assert pha.get_dep(filter=False) == pytest.approx([1, 2, 0, 3, 12])
+    assert pha.get_dep(filter=True) == pytest.approx([3])
+
+    # Change the grouping; it would be nice if it could have
+    # recognized the requested range was > 1 and <= 5 but the current
+    # code does not support this.
+    #
+    pha.grouping = [1] * 5
+
+    assert pha.get_dep(filter=False) == pytest.approx([1, 2, 0, 3, 12])
+    assert pha.get_dep(filter=True) == pytest.approx([3])
+
+    assert pha.get_filter() == "4"
+
+
+def test_pha_grouping_changed_1160_ungrped_with_filter(make_grouped_pha):
+    """Test based on work on #1160
+
+    A version of
+    test_pha_grouping_changed_1160_grped_with_filter
+    but the data is not grouped, even though the grouping
+    is set/changed.
+
+    """
+
+    pha = make_grouped_pha
+
+    # Apply the filter whilst still grouped
+    pha.ignore(hi=1)
+    pha.ignore(lo=5)
+
+    pha.ungroup()
+
+    # The filtering does not change because of the ungroup call,
+    # although we might like it too.
+    assert pha.get_filter() == "4"
+
+    assert pha.get_dep(filter=False) == pytest.approx([1, 2, 0, 3, 12])
+    assert pha.get_dep(filter=True) == pytest.approx([3])
+
+    # Change the grouping
+    pha.grouping = [1] * 5
+
+    assert pha.get_dep(filter=False) == pytest.approx([1, 2, 0, 3, 12])
+    assert pha.get_dep(filter=True) == pytest.approx([3])
+
+    assert pha.get_filter() == "4"
+
+
+@requires_fits
+@requires_data
+def test_1160(make_data_path):
+    """Use the dataset we reported this with just as an extra check
+
+    It is slightly different to the other #1160 tests above because
+
+    a) we do not have a non-zero quality bin
+    b) we have an instrument response (this should not really
+       matter here)
+
+    """
+
+    from sherpa.astro.io import read_pha
+    pha = read_pha(make_data_path("3c273.pi"))
+
+    fexpr = "0.47:6.57"
+
+    pha.notice(0.5, 6)
+    assert pha.get_dep(filter=False).shape == (1024, )
+    assert pha.get_dep(filter=True).shape == (41, )
+    assert pha.mask.shape == (46, )
+    assert pha.mask.sum() == 41
+    assert pha.get_filter(format="%.2f") == fexpr
+
+    pha.grouping = [1] * 1024
+    assert pha.get_dep(filter=False).shape == (1024, )
+    # This fails with a DataErr: size mismatch between mask and data array
```

Comment:
What is "this" that fails? The test passes. Do you mean "should fail"?

Reply:
It could be a line that should have been removed. I'll have to review.

Reply:
This test is marked xfail in cd2569d2 and so the text just hasn't been updated when the code was fixed.

Reply:
Oops - that's the wrong commit. It should be the second commit added as part of this PR.